pyiceberg/table/__init__.py

```diff
@@ -3537,6 +3537,39 @@ def update_partitions_map(
             schema=table_schema,
         )
 
+    def metadata_log_entries(self) -> "pa.Table":
+        import pyarrow as pa
+
+        from pyiceberg.table.snapshots import MetadataLogEntry
+
+        table_schema = pa.schema([
+            ("timestamp", pa.timestamp(unit='ms'), True),
+            ("file", pa.string(), True),
```

Comment:
why are file and timestamp nullable?  In what scenarios is it expected to have a log entry with a snapshot id but no timestamp or file?

Reply:
ah, good catch. `timestamp` and `file` should both be required fields, according to the [Java schema](https://github.com/apache/iceberg/blob/1e35bf96ecacd5c5175116f40fa3e097991d04d2/core/src/main/java/org/apache/iceberg/MetadataLogEntriesTable.java#L30-L35). The third element of the tuple represents [`nullable`](https://arrow.apache.org/docs/python/generated/pyarrow.schema.html), which should be `False` for both.

Reply:
This means the rest of the field's nullable fields are also wrong.